src/marqo/core/index_management/index_management.py

```diff
@@ -198,11 +202,18 @@ def update_index(self, marqo_index: SemiStructuredMarqoIndex) -> None:
             OperationConflictError: If another index creation/deletion operation is
                 in progress and the lock cannot be acquired
         """
-        if not isinstance(marqo_index, SemiStructuredMarqoIndex):
-            # This is just a sanity check, it should not happen since we do not expose this method to end user.
-            raise InternalError(f'Index {marqo_index.name} can not be updated.')
-
         with self._vespa_deployment_lock():
+            existing_index = self.get_index(marqo_index.name)
+            if not isinstance(existing_index, SemiStructuredMarqoIndex):
+                # This is just a sanity check, it should not happen since we do not expose this method to end user.
+                raise InternalError(f'Index {marqo_index.name} created by Marqo version {marqo_index.marqo_version} '
+                                    f'can not be updated.')
+
+            if (existing_index.tensor_field_map == marqo_index.tensor_field_map and
```

Comment:
The other thread may have added other things too, but also your field. An exact match will miss that

Reply:
Changed the logic so it only updates the index when there's fields missing.